Static Swagger Documentation

nguyenfilip opened the pull request:
Created project folder 'swagger' that uses an empty 'swagger' layout. This way,
the Swagger UI is served statically. Swagger descriptor (swagger.json) is the
input for the static documentation. It has to be pushed into repository when we
want to update the documentation. The swagger.json files for Candlepin will
reside in swagger/candlepin folder. We can keep multiple Swagger files for
different versions of Candlepin  in there and modify links to Swagger UI so that
the link shows correct version of the documentation.

I also deprecated 5.1 Rest API and created a new Swagger API page with
information about runtime behavior of Swagger integration and links to static
Swagger documentation.

nguyenfilip commented:
@awood This is still in progress until the main Swagger PR is merged, but could you let me know what do you think about this way of publishing static Swagger doc? Maybe you know about a more appropriate way.

awood commented on docs/candlepin/swaggerapi.md:
```diff
@@ -0,0 +1,23 @@
+
+| Version | Swagger file | Interactive documentation |
+|-
+| master  |  [swagger.json]({{ site.url }}/swagger/candlepin/swagger.json)  | [Swagger UI]({{ site.url }}/swagger) |
```
Be careful with internal links.  Preface them with {{ site.baseurl }} if they are in another directory.  See
http://jekyllrb.com/docs/github-pages/#project_page_url_structure

awood commented:
Let's shrink this a little by dropping all the translation JS files.  All of the actual documentation is in English, so I don't see much utility for non-English speaking users in having a handful of terms in the interface in their native language.

nguyenfilip commented:
@awood Yeah thats a good suggestion! Removed the translations! 

Thanks for reviewing this ahead of time! I will notify you when the swagger support is merged to the master so that we can merge this.

nguyenfilip commented:
@awood Ok this is ready for final review. I amended a few more things into this:

- upgraded swagger-ui.js because of the bugs that has been fixed in the new version
- added swagger.json for Candlepin 2.0.13
- fixed links from static documentation, so that the interactive documentation now opens correctly
- added API sorting of resources and operations

awood commented on docs/candlepin/api.html:
```diff
@@ -2,6 +2,7 @@
 title: REST API
 additional_json: candlepin_methods.json
 ---
+This REST API is deprecated since Candlepin now supports Swagger API. See [here](swaggerapi)
```
This page is html, not markdown, so you'll need to put the link in an anchor tag.

On a more general note, I think it would be find to just delete api.html entirely as part of this PR.

nguyenfilip replied:
removed api.html ;-)

awood commented:
You'll probably need to coordinate some with @wottop and let him know what he needs to do now at the end of sprint to update the API docs.

awood commented:
Looks great on the site though.  As far as updating the API every sprint: we could write a Jekyll plugin to just introspect the `swagger/candlepin` directory and generate the `swaggerapi.md` page for us.

nguyenfilip commented:
@wottop At the end of each sprint it is necessary to download the swagger.json file into docs/swagger/candlepin. Name it based on the version (e.g. swagger-2.0.13.json) and then add a new link to the docs/docs/candlepin/swaggerapi.md to that file. Are there some end-of-sprint notes, that you would like me to update?
